app/src/main/java/com/example/checknodeavailability/MainActivity.kt

```diff
@@ -8,30 +9,55 @@ import com.jcraft.jsch.JSch
 import android.os.AsyncTask
 import android.util.Log
 import android.widget.Button
+import android.widget.EditText
 import java.io.ByteArrayOutputStream
+import java.io.FileOutputStream
 import java.io.InputStream
 import java.lang.Exception
 
 
 class MainActivity : AppCompatActivity() {
-    val user = System.getenv("username") ?: ""
-    val password = System.getenv("password") ?: ""
-    val host = System.getenv("host") ?: ""
 
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
         setContentView(R.layout.activity_main)
+        val usernameField = findViewById<EditText>(R.id.username)
+        val IPAddressField = findViewById<EditText>(R.id.ipAddress)
+        val passwordField = findViewById<EditText>(R.id.password)
+        val fileOutputStream:FileOutputStream
+
+        val filename = "logininfo.txt"
+
+
+
         val button = findViewById<Button>(R.id.button)
         button.setOnClickListener {
-            ExecuteSSHCommand().execute()
+            val username = usernameField.getText().toString()
```

Comment:
In Kotlin, you don't need to use getters and setters, they're in-built
So you should be able to just write `usernameField.text` directly